### PR TITLE
[REF] mail: move various channel code to discuss folder

### PR DESCRIPTION
**Pull request by seb-odoo**
Part of task-3265211

Note: PR includes https://github.com/odoo/odoo/pull/125397 adapted to master code

### CODE REVIEW

File: addons/mail/static/src/core/common/chat_window_service.js [hunk 1]
```diff
@@ -24,6 +24,19 @@ export class ChatWindowService {
         this.ui = services.ui;
     }
 
+    open(thread, replaceNewMessageChatWindow) {
```
Comment (phenix-factory) [inline addons/mail/static/src/core/common/chat_window_service.js:27]: We could @type this function since there is diff on it now.

**Review by alexkuhn — APPROVED** [review]
LGTM

File: addons/mail/static/src/core/common/chat_window_service.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/core/common/chat_window_service.js:27]: I'd do if I had other changes to do, but as it's the only minor comment I prefer if the PR moves forward.

We can always make follow up to add missing types, we don't need to wait for diff on the methods to do it.

**Review by seb-odoo — COMMENTED** [review]
@robodoo r+

**Comment by seb-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**